lib/csv_schema.ex

```diff
@@ -66,83 +79,110 @@ defmodule Csv.Schema do
   the relative path to csv file in your project. Second parameter should be a `field` list
   included in `do`-`end` block
   """
-  defmacro schema(file_path, do: block) do
+  defmacro schema([string: string], do: block) do
+    quote do
+      to_stream = fn -> unquote(string) |> String.split("\n") |> Enum.reject(&(&1 == "")) |> Stream.map(& &1) end
+      Module.put_attribute(__MODULE__, :to_stream, to_stream)
+      unquote(__register__(block))
+      unquote(__explode__())
+    end
+  end
+
+  defmacro schema([path: file_path], do: block) do
+    quote do
+      Module.put_attribute(__MODULE__, :external_resource, unquote(file_path))
+      Module.put_attribute(__MODULE__, :to_stream, fn -> File.stream!(unquote(file_path)) end)
+      unquote(__register__(block))
+      unquote(__explode__())
+    end
+  end
+
+  # DEPRECATED. Consider using schema(list(content: String.t() | path: String.t(), do: block) instead
+  defmacro schema(file_path, do: block) when is_binary(file_path) do
```

Comment:
If we cannot make it backward-compatible I would use the @deprecated module attribute

Reply:
The problem is that `@deprecated` work with function name and arity. It means that the other schema matches if used would warn in console at compile time

Reply:
Ok, so we could just leave the comment. As an alternative, instead of making the first argument a keyword list, we could leave the first argument as a binary, and then specify with a boolean option (in keyword list) if we are passing a file path or the data directly, something like this:
```elixir
schema "path/to/csv_file.csv", file_path: true do
  ..
end
```
and give a default of true to the option.

This way it would be backward compatible with no deprecation needed.
and
```elixir
schema csv_file_content, file_path: false do
  ..
end
```